main.go

```diff
@@ -19,27 +20,68 @@ func determineListenPort() (string, error) {
     return ":" + port, nil
 }
 
-func handleGetRootRequest(
+func respondWithMethodNotAllowed(
+        responseWriter http.ResponseWriter,
+        allowedMethods []string) {
+
+    statusCode := http.StatusMethodNotAllowed
+
+    responseWriter.Header().Set(
+        "Allow",
+        strings.Join(allowedMethods, ", "))
+
+    http.Error(
+        responseWriter,
+        http.StatusText(statusCode),
+        statusCode)
+}
+
+func handleLoginOrSignupRequest(
         responseWriter http.ResponseWriter,
         request *http.Request) {
 
-    parsedTemplate, err := template.ParseFiles("root.tmpl")
-    if err != nil {
-        log.Fatal(err)
-    }
+    switch request.Method {
+        case http.MethodGet:
+            parsedTemplate, err := template.ParseFiles("templates/login_or_signup.tmpl")
+            if err != nil {
+                log.Fatal(err)
+            }
 
-    parsedTemplate.Execute(responseWriter, nil)
+            parsedTemplate.Execute(responseWriter, nil)
+
+        default:
+            respondWithMethodNotAllowed(
+                responseWriter,
+                []string{http.MethodGet})
+    }
 }
 
 func main() {
+    // SET ROUTER
+
+    // static files
+    {
```

Comment:
This block is just for handling static assets? So anything we request that is static will automatically be redirected and go will look for things in `./static/` folder? maybe add a comment for the go uninitiated like me

Reply:
I wouldn't say "redirected", but yeah your understanding is correct. This was the standard way to treat static files in Django as well. What comment would have helped you understand this code quicker?

Reply:
its probably cause i'm not used to it, but static file handler boilerplate is what i would have put it.